### ISSUE TITLE
New Module WebHook with Slack

### ISSUE THREAD

**Issue by josett225** [Dolibarr/dolibarr#21359]
### Bug

@Eldy and @Hystepik 

Many thanks for the WebHook module, I just tried to make it work with Slack with trigger COMPANY_MODIFY. The debug shows that the trigger works but I got initially an error 400. I believe after reviewing the code that the function getURLContent() does not support the body (--data) feature :
Here is the slackcall : curl -X POST -H 'Content-type: application/json' --data '{"text":"Hello, World!"}' https://hooks.slack.com/services/................

I did some changes in your trigger code instead of using getURLContent () and it works now. I know that supporting slack messaging is a full module development but it was for testing purpose.

Here is the code used instead of using getURLContent in interface_99_modWebhook_WebhookTriggers.class.php:
`
$curl = curl_init();

curl_setopt_array($curl, array(
  CURLOPT_URL => 'https://hooks.slack.com/services/blabla/blabla2/secret',
  CURLOPT_RETURNTRANSFER => true,
  CURLOPT_ENCODING => '',
  CURLOPT_MAXREDIRS => 10,
  CURLOPT_TIMEOUT => 0,
  CURLOPT_FOLLOWLOCATION => true,
  CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
  CURLOPT_CUSTOMREQUEST => 'POST',
  CURLOPT_POSTFIELDS => '{"text":'.json_encode($action).'}';,
  CURLOPT_HTTPHEADER => array(
    'Content-Type: application/json'
  ),
));

$response = curl_exec($curl);

curl_close($curl);
`

Let me know if you need mode information.

### Environment Version
Dolibarr Version 16 Beta

### Environment OS
MAMP

### Environment PHP
7.4

### Environment Database
MYSQL

**Comment by josett225** [comment]
To be tested with v17 and v18

**Comment by ksar-ksar** [comment]
Hello,

Is it still the case on V18 ?

**Comment by josett225** [comment]
Hi @ksar-ksar 
Yes there is still 2 issues/bugs and a feature request here. 
1 unable to define the type of post request. 
2 if webhook activated with even the target(s) deactivated, any trigger declared is blocking the action. Example company_create due to type of request and payload content. 
3 feature request 
Payload editor with Dolibarr placeholder (as used in Emails)

For testing purposes, I corrected all the issues above to make it work.
For issue 3 I used and redirect the description entry of target to enter a slack payload 
I am still testing and cleaning code. 
So I will share later.
